### PR TITLE
IPL Winner Prediction

### PR DESCRIPTION
**Pull request by Soumayan-pal01**
## IPL Winner Prediction 

- The aim of this project is to perform analysis of the IPL data and predict the winner of the IPL matches.

- [x] LGM-SOC'21 Participant 
- [ ] DevIncept Participant
- [ ] Contributor

Closes: #186

### Describe the add-ons or changes you've made

In this project we have performed a detailed analysis and visualization of the training dataset with different Exploratory Data Analysis techniques. Then a comaparative analysis of different ML Classifiers have been done to predict the winners of the IPL matches. After performing the model comparative analysis we can conclude that only the Random Forest Classifier and XGBoost Classifier have performed well. Hypertuning using RandomisedSearchCV anD GridSearchCV have also been done to prevent overfitting and the best accuracy achieved is 66.4%

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by prathimacode-hub** [comment]
Why have you closed this? @Soumayan-pal01

**Comment by Soumayan-pal01** [comment]
## IPL Winner Prediction 

- The aim of this project is to perform analysis of the IPL data and predict the winner of the IPL matches.

- [x] LGM-SOC'21 Participant 
- [ ] DevIncept Participant
- [ ] Contributor

Closes: #297 

### Describe the add-ons or changes you've made

In this project we have performed a detailed analysis and visualization of the training dataset with different Exploratory Data Analysis techniques. Then a comaparative analysis of different ML Classifiers have been done to predict the winners of the IPL matches. After performing the model comparative analysis we can conclude that only the Random Forest Classifier and XGBoost Classifier have performed well. Hypertuning using RandomisedSearchCV anD GridSearchCV have also been done to prevent overfitting and the best accuracy achieved is 66.4%

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update


## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

**Comment by Soumayan-pal01** [comment]
I gave the wrong issue tag actually @prathimacode-hub

**Comment by Soumayan-pal01** [comment]
Creating a fresh one.

**Comment by prathimacode-hub** [comment]
You can edit it right instead of closing it. @Soumayan-pal01

**Comment by Soumayan-pal01** [comment]
Ohh..I forgot about. Anyway I have created a new one
